src/workflows/airqo_etl_utils/storage/cloud_storage.py

```
@@ -0,0 +1,448 @@
+            with NamedTemporaryFile(delete=False, suffix=f".{format}") as temp:
+                if format == "csv":
+                    dataframe.to_csv(temp.name, index=False)
+                    mimetype = "text/csv"
+                elif format == "json":
+                    dataframe.to_json(temp.name, orient="records")
+                    mimetype = "application/json"
+                elif format == "parquet":
+                    dataframe.to_parquet(temp.name)
+                    mimetype = "application/octet-stream"
+                else:
+                    raise ValueError(f"Unsupported format: {format}")
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)